### PR TITLE
chore(tests): Migrate from Enzyme to RTL (1st batch)

### PR DESCRIPTION
**Pull request by loonskai**
### Description

Tests with Enzyme migrated: **23/61**

This PR is a part of Box-wide migration to the newer React v18. 
- The most of the scenarios were rewritten with consideration of RTL practices (thanks @nmkedziora for pointing this out). Some components required `fireEvent` executed on top of specific selectors instead of recommended `userEvent`. Some of these tradeoffs could be avoided (possibly) but would extend the timelines of this PR. I suggest improve these test iteratively as we work on them.
- Semantic selectors were used in most cases. `data-testid` attributes were added mostly for container elements to quickly validate if container is rendered.
- This PR doesn't introduce any runtime changes (except `data-testid` changes mentioned above)

This PR will be followed with the next steps:
- Rewrite the rest of test scenarios from Enzyme to RTL and cleanup Enzyme-related setup (dependencies, adapter etc.)
- Add [eslint-plugin-testing-library](https://github.com/testing-library/eslint-plugin-testing-library) and [eslint-plugin-jest-dom](https://github.com/testing-library/eslint-plugin-jest-dom) as ESLint rules

### Screenshots
image

### CODE REVIEW

File: src/lib/viewers/box3d/image360/__tests__/Image360Controls-test.tsx
Comment (nmkedziora) [inline src/lib/viewers/box3d/image360/__tests__/Image360Controls-test.tsx:13]: Do not use `wrapper` concept when working with React Testing Library. 

> _The name `wrapper` is old cruft from enzyme and we don't need that here. The return value from `render` is not "wrapping" anything. It's simply a collection of utilities that (thanks to the next thing) you should actually not often need anyway._

Source:[ Common mistakes with React Testing Library](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-wrapper-as-the-variable-name-for-the-return-value-from-render) by Kent C.Dodds

I highly recommend to get familiar with this article.

File: src/lib/viewers/box3d/image360/__tests__/Image360Controls-test.tsx
Comment (nmkedziora) [inline src/lib/viewers/box3d/image360/__tests__/Image360Controls-test.tsx:25]: Ditto. You do not need wrapper to find elements that were rendered - use `screen` instead. 
Also, I think all the `await` could probably omitted.

Source: [Not using screen](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-screen)

File: src/lib/viewers/box3d/image360/__tests__/Image360Controls-test.tsx
Comment (nmkedziora) [inline src/lib/viewers/box3d/image360/__tests__/Image360Controls-test.tsx:27]: What is `click()`?
Recommended way of working with events is to use `fireEvent` (a few use cases) and `userEvent` (majority of the time) [@testing-library/user-event](https://testing-library.com/docs/user-event/intro).

Source: [Firing Events at Testing Library docs](https://testing-library.com/docs/dom-testing-library/api-events/)

File: src/lib/viewers/box3d/image360/__tests__/Image360Controls-test.tsx
Comment (nmkedziora) [reply src/lib/viewers/box3d/image360/__tests__/Image360Controls-test.tsx:25]: > _The base queries from DOM Testing Library require you to pass a container as the first argument. Most framework-implementations of Testing Library provide a pre-bound version of these queries when you render your components with them which means you do not have to provide a container. In addition, if you just want to query document.body then you can use the [screen](https://testing-library.com/docs/queries/about/#screen) export as demonstrated below (**using screen is recommended**)._

Source: [Using Queries](https://testing-library.com/docs/queries/about/#using-queries)

File: src/lib/viewers/archive/__tests__/ArchiveExplorer-test-react.js
Comment (PaQ666) [inline src/lib/viewers/archive/__tests__/ArchiveExplorer-test-react.js:110]: :thought_balloon: Isn't it enough to just search for this element in the document/screen which would fail if it does't exist?

File: src/lib/viewers/box3d/model3d/__tests__/Model3DControlsNew-test.tsx
Comment (PaQ666) [inline src/lib/viewers/box3d/model3d/__tests__/Model3DControlsNew-test.tsx:73]: Tool creators suggest to initialise userEvent before tests: https://testing-library.com/docs/user-event/intro
Not sure if relevant, just noticed it while checking docs.

File: src/lib/viewers/controls/settings/__tests__/SettingsList-test.tsx
Comment (rafalmaksymiuk) [inline src/lib/viewers/controls/settings/__tests__/SettingsList-test.tsx:64]: Isn't it redundant? I mean in DOM only one element can have focus?

File: src/lib/viewers/controls/settings/__tests__/SettingsDropdown-test.tsx
Comment (rafalmaksymiuk) [inline src/lib/viewers/controls/settings/__tests__/SettingsDropdown-test.tsx:126]: Maybe to properly test it we should actual user event like click that focuses on element as well? 
If you await for such event then `waitFor` might not be needed?

**Comment by rafalmaksymiuk** [comment]
This is a massive PR - tremendous work but hard to review. I know it is difficult to push each PR separately, but is way easier to meaningfully review so maybe split more next time :)?

Another thought - or rather question. I do know that EUA uses special tooling based on RTL. Don't remember details, but it integrates Redux somehow. Do you know why it was needed there and if it might be here?

**Review by rafalmaksymiuk — DISMISSED** [review]

File: src/lib/viewers/archive/__tests__/ArchiveExplorer-test-react.js
Comment (nmkedziora) [reply src/lib/viewers/archive/__tests__/ArchiveExplorer-test-react.js:110]: Do you mean this 👇  @rafalmaksymiuk ?

```suggestion
            expect(screen.getByText('test.zip')).toBeInTheDocument();
```

If yes, I think this shortened version is much more readable.

File: src/lib/viewers/controls/settings/__tests__/SettingsDropdown-test.tsx
Comment (nmkedziora) [reply src/lib/viewers/controls/settings/__tests__/SettingsDropdown-test.tsx:126]: I agree. Currently you test (native) focus implementation.
Instead, I'd do sth like in this [example](https://testing-library.com/docs/user-event/convenience/): 

```
 test('should be able to focus on the dropdown button', async() => {
            const user = userEvent.setup()
            renderView();

            const button = screen.getByRole('button', { name: /first/i });

           await user.tab() 

           expect(button).toHaveFocus();
        });
```

**Comment by nmkedziora** [comment]
> This is a massive PR - tremendous work but hard to review. I know it is difficult to push each PR separately, but is way easier to meaningfully review so maybe split more next time :)?

+1 for me. 
Amazing job and great effort 🎉 

Splitting the PR by feature or folder would make a significant difference. It would not only make it easier to review but also quicker to merge bit by bit instead of blocking the merge for the entire PR due to comments on a single file.

File: src/lib/viewers/archive/__tests__/Breadcrumbs-test-react.js
Comment (nmkedziora) [inline src/lib/viewers/archive/__tests__/Breadcrumbs-test-react.js:7]: Nit: It's a small and simple render, so I think using `render` directly would be more readable. I highly recommend not using helper functions like that whenever possible or trying to name the helper function in a more meaningful manner, i.e. `renderBreadcrumb`. `renderView` is too generic IMHO, and because of that I need to check the helper implementation in order to understand the test itself.

File: src/lib/viewers/archive/__tests__/Breadcrumbs-test-react.js
Comment (nmkedziora) [inline src/lib/viewers/archive/__tests__/Breadcrumbs-test-react.js:28]: Nit: you could do `screen.getAllByRole('button')` only once:
```suggestion
            const buttons = screen.getAllByRole('button')
            expect(buttons.at(1).textContent).toContain('test');
```

File: src/lib/viewers/archive/__tests__/ArchiveExplorer-test-react.js
Comment (rafalmaksymiuk) [reply src/lib/viewers/archive/__tests__/ArchiveExplorer-test-react.js:110]: Rather simple `within(screen.getByLabelText('Breadcrumb')).getByText('test.zip')` which would serve same purpose, but then there is good argument here about the intention:
https://stackoverflow.com/questions/73281448/why-we-have-to-use-tobeinthedocument-in-testing-library

Comment (loonskai) [reply src/lib/viewers/archive/__tests__/ArchiveExplorer-test-react.js:110]: @rafalmaksymiuk yes, I referred to the same recommendation

**Review by rafalmaksymiuk — APPROVED** [review]

File: src/lib/viewers/media/__tests__/DashControls-test.tsx
Comment (bkepka-box) [inline src/lib/viewers/media/__tests__/DashControls-test.tsx:62]: Should we extract it in order to not duplicate setting props in every test case?

**Review by bkepka-box — APPROVED** [review]

**Pull request merged**
